plugins/schedule/base/lib/model/ScheduleEventPeer.php

```diff
@@ -249,6 +251,30 @@ public static function retrieveEventsByResourceIdsAndDateWindow($resourceIds, $s
 
 		return self::doSelect($c);
 	}
+
+	/**
+	 * @param date $startDate
+	 * @param date $endDate
+	 * @param string|null $scheduleEventIdToIgnore
+	 * @return array <ScheduleEvent>
+	 */
+	public static function retrieveBlackoutEventsByDateWindow($startDate, $endDate, $scheduleEventIdToIgnore = null)
```

Comment:
Since this function and the one above are 95% the same, it would be better to have one function that each over laps it and it is variate to the criteria - 
getEventCriteiaByDateWindow() that will return criteria , and each function will only add it variant of type.